src/scaffoldmaker/scaffoldpackage.py

```diff
@@ -75,12 +92,116 @@ def getScaffoldSettings(self):
     def getScaffoldType(self):
         return self._scaffoldType
 
-    def generate(self, region):
+    def getRotation(self):
+        return self._rotation
+
+    def setRotation(self, rotation):
+        '''
+        :param rotation: list of 3 rotation angles about z, y', x'', in degrees.
+        :return: True if value changed, otherwise False.
+        '''
+        assert len(rotation) == 3
+        if self._rotation != rotation:
+            self._rotation = rotation
+            return True
+        return False
+
+    def getScale(self):
+        return self._scale
+
+    def setScale(self, scale):
+        '''
+        :param scale: list of 3 scales for x, y, z.
+        :return: True if value changed, otherwise False.
+        '''
+        assert len(scale) == 3
+        if self._scale != scale:
+            self._scale = scale
+            return True
+        return False
+
+    def getTranslation(self):
+        return self._translation
+
+    def setTranslation(self, translation):
+        '''
+        :param translation: list of 3 translation values for x, y, z.
+        :return: True if value changed, otherwise False.
+        '''
+        assert len(translation) == 3
+        if self._translation != translation:
+            self._translation = translation
+            return True
+        return False
+
+    def getTransformationMatrix(self):
+        '''
+        :return: 4x4 row-major transformation matrix with first index down rows, second across columns,
+        suitable for multiplication p' = Mp where p = [ x, y, Z, h ], or None if identity.
+        '''
+        # apply transformation in order: scale then rotation then translation
+        if not all((v == 0.0) for v in self._rotation):
+            rotationMatrix = vectorops.eulerToRotationMatrix3([ deg*math.pi/180.0 for deg in self._rotation ])
+            return [
+                [ rotationMatrix[0][0]*self._scale[0], rotationMatrix[0][1]*self._scale[1], rotationMatrix[0][2]*self._scale[2], self._translation[0] ],
+                [ rotationMatrix[1][0]*self._scale[0], rotationMatrix[1][1]*self._scale[1], rotationMatrix[1][2]*self._scale[2], self._translation[1] ],
+                [ rotationMatrix[2][0]*self._scale[0], rotationMatrix[2][1]*self._scale[1], rotationMatrix[2][2]*self._scale[2], self._translation[2] ],
+                [ 0.0, 0.0, 0.0, 1.0 ] ]
+        if not (all((v == 1.0) for v in self._scale) and all((v == 0.0) for v in self._translation)):
+            return [
+                [ self._scale[0], 0.0, 0.0, self._translation[0] ],
+                [ 0.0, self._scale[1], 0.0, self._translation[1] ],
+                [ 0.0, 0.0, self._scale[2], self._translation[2] ],
+                [ 0.0, 0.0, 0.0, 1.0 ] ]
+        return None
```

Comment:
This may not be relevant, but was wondering if scaling would still work as before if rotation and translation are both 0 but the scale is changed?

Reply:
It should be handled by the second case, just with zeros in the appropriate column.